### ISSUE TITLE
New flag to change default umask?

### ISSUE THREAD

**Issue by iyanmv** [jupyter/docker-stacks#700]
Hi all,

I use Jupyter images very ofter at work. I always set NB_UID and NB_GID env variables to keep notebooks and code with the same owner and group as in the host machine. This is very useful when we need to access that code by other apps in the server.

In some cases, I also found useful to mount additional folders (apart from `/home/jovyan/work`) (e.g. sharing folder between a group of scientists). The problem is that default umask (both in bash and jupyter) does not allow to modify files creates by other user even in the same group.

So imagine this situation. Two people, *Alice* and *Bob* have their own jupyter notebook set up with their UIDs and a common GID. Both have a mounted volume in `/home/jovyan/shared_folder`. When *Alice* creates a new file in that folder, *Bob* can read it but he cannot modify it. Changing umask from 0022 to 0002 (or 0007) solves this issue.

Do you think having an extra flag to set up default umask would be helpful? If you think so, I can create a pull request. Maybe NB_UMASK?

Best regards,
Iyán

**Comment by rkdarst** [comment]
+1, and in fact I recently did this locally.  Actually not by modifying start.sh, but by modifying the docker run command to hack it before control got to the start.sh script.

For me, it was difficult because I was using root running + the internal sudo, and there were a surprisingly number of difficulties.  Here is what all I had to do, in the end:

```
umask 0007
sed -r -i 's/^(UMASK.*)022/\1007/' /etc/login.defs
echo Defaults umask=0007, umask_override >> /etc/sudoers
```
I think the second one may not be needed, I did many things and before I found a working solution.  If you have a patch, I can try to test it and use it one it is ready.


disclaimer: not involved in docker-stacks project
